civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -46,17 +44,14 @@
          * @param {string} mode
          */
         function modalDocument (id, role, mode) {
-          $rootScope.$broadcast('ct-spinner-show', 'document-' + id);
-          vm.loadingModalData = true;
-
           DocumentService.get({ id: id })
             .then(function(data) {
               if (!data) {
                 throw new Error('Requested Document is not available');
               }
-
+              vm.openModalDocument(data[0], role, mode);
```

Comment:
if you make `openModalDocument` return the promise of `modalInstance.opened.then()`, then you can write a more expressive code

```js
vm.openModalDocument(data[0], role, mode)
  .then(function () {
    return vm.cacheContact(data);
  })
  .then(function () {
    $rootScope.isLoading = false;
  });
```

Reply:
Even better, i think
```js
  DocumentService.get({ id: id })
    .then(function (data) {
      return data;  // or if not present, throw an exception
    })
    .then(function (data) {
      return vm.openModalDocument();
    })
    // etc..
    .catch(function (reason) {
      // the code should go here even if the first `then` throws an error, please double check
    });
```